[Docs] Updated Read Me (#3077)
m3db/m3: README.md

    "value": 3347.26
}'
```

5.  Query Results

**Linux**

```shell
curl -X "POST" -G "http://localhost:7201/api/v1/query_range" \
  -d "query=third_avenue" \
  -d "start=$(date "+%s" -d "45 seconds ago")" \
  -d "end=$( date +%s )" \
  -d "step=5s" | jq .  
```

**macOS/BSD**

```shell
curl -X "POST" -G "http://localhost:7201/api/v1/query_range" \
  -d "query=third_avenue > 6000" \
  -d "start=$(date -v -45S "+%s")" \
  -d "end=$( date +%s )" \
  -d "step=5s" | jq .
```

## Contributing

You can ask questions and give feedback in the following ways:

-   [Create a GitHub issue](https://github.com/m3db/m3/issues)
-   [In the public M3 Slack](http://bit.ly/m3slack)
-   [In the M3 forum (Google Group)](https://groups.google.com/forum/#!forum/m3db)

M3 welcomes pull requests…
Read [the development guide](DEVELOPMENT.md) to help you get setup for building and contributing to M3.

## Community Meetings

M3 contributors and maintainers have regular meetings. Join our M3 meetup group to receive notifications on upcoming meetings: 
<https://www.meetup.com/M3-Community/>.

Recordings of past meetups can be found here: <https://vimeo.com/user/120001164/folder/2290331>.

> ## Office Hours

Reviewer: Only thing I would change is to move these three sections to be under "More information".

i.e. headers order should be
1. More information
2. Community meetings
3. Office hours
4. ...everything else...
5. Contributing
6. "This project is released under Apache License, Version 2.0"... etc

This matches the fairly popular as in use with etcd and other projects:
https://github.com/etcd-io/etcd

Reviewer 2: @robskillington We were also following a very popular read me template ;)

https://github.com/RichardLitt/standard-readme

Reviewer 2: but let me check, it's slightly hard to know which sections you're referring to as the diff is so big

Reviewer 2: Though that template doesn't really have a "more information" section

Reviewer 2: OK, think I updated to what you mean, I was mostly concerned it would push the "getting started" steps too far down, but really we want readers to go to the docs anyway

[Distributed TSDB](https://docs.m3db.io/m3db/) and [Query Engine](https://docs.m3db.io/how_to/query/), [Prometheus Sidecar](https://docs.m3db.io/integrations/prometheus/), [Metrics Aggregator](https://docs.m3db.io/overview/components/#m3-aggregator), and more such as [Graphite storage and query engine](https://docs.m3db.io/integrations/graphite/).

> Table of Contents:

Reviewer 2: ```suggestion
## Table of Contents:
```

> - [Install](## Install)

Reviewer 2: ```suggestion
- [Install](#install)
```

> - [Dependencies](## Dependencies)

Reviewer 2: ```suggestion
- [Dependencies](#dependencies)
```

> - [Usage](## Usage)

Reviewer 2: ```suggestion
- [Usage](#usage)
```

> - [Contributing](## Contributing)

Reviewer 2: ```suggestion
- [Contributing](#contributing)
```

> - [Community meetings](## Community Meetings)

Reviewer 2: ```suggestion
- [Community meetings](#community-meetings)
```

> - [Office Hours](## Office Hours)

Reviewer 2: ```suggestion
- [Office Hours](#office-hours)
```

> More information: 

Reviewer 2: ```suggestion
## More information
```

This actually was't in our Doc…

- [Documentation](https://docs.m3db.io/)
- [Developer Guide: Getting Started](https://github.com/m3db/m3/blob/master/DEVELOPMENT.md)
- [Slack](http://bit.ly/m3slack)
- [Forum (Google Group)](https://groups.google.com/forum/#!forum/m3db)

## Install 

> ## Dependencies 

Reviewer 2: ```suggestion
### Dependencies 
```

Probably makes sense

> The simplest and quickest way to try M3 is to use Docker, read the M3 getting started section for other options. 

Reviewer 2: ```suggestion
The simplest and quickest way to try M3 is to use Docker, read [the M3 quickstart section](https://m3db.io/docs/quickstart) for other options. 
```

> ```

Reviewer 2: Actually we can scrap this now…

```suggestion

```

> docker pull quay.io/m3db/m3dbnode:latest

Reviewer 2: ```suggestion
```

> ```

Reviewer 2: ```suggestion
```

## Usage 

The below is a simplified version of the [M3 quickstart guide](https://m3db.io/docs/quickstart/docker/), and we suggest you read that for more details.

1. Start a container

> ```

Reviewer 2: ```suggestion
```shell
```

docker run -p 7201:7201 -p 7203:7203 --name m3db -v $(pwd)/m3db_data:/var/lib/m3db quay.io/m3db/m3dbnode:v1.0.0
```

2. Create a placement and namespace

> ```

Reviewer 2: ```suggestion
```shell
```

#!/bin/bash
curl -X POST http://localhost:7201/api/v1/database/create -d '{
  "type": "local",
  "namespaceName": "default",
  "retentionTime": "12h"
}' | jq .
```

3. Ready a namespace

> ```

Reviewer 2: ```suggestion
```shell
```

curl -X POST http://localhost:7201/api/v1/services/m3db/namespace/ready -d '{
  "name": "default"
}' | jq .
```

4. Write metrics

> ```

Reviewer 2: ```suggestion
```shell
```

#!/bin/bash
curl -X POST http://localhost:7201/api/v1/json/write -d '{
  "tags": 
    {
      "__name__": "third_avenue",
      "city": "new_york",
      "checkout": "1"
    },
    "timestamp": '\"$(date "+%s")\"',
    "value": 3347.26
}'
```

> 5. Query results

Reviewer 2: There's a slight issue with these commands thanks to the different ways that OSs handle dates. So actually we need to add two examples here annoyingly.

> ```

Reviewer 2: ```suggestion
```shell
```

curl -X "POST" -G "http://localhost:7201/api/v1/query_range" \
  -d "query=third_avenue" \
  -d "start=$(date "+%s" -d "45 seconds ago")" \
  -d "end=$( date +%s )" \
  -d "step=5s" | jq .  
```

> 

Reviewer 2: ```suggestion
**macOS/BSD**
```shell
curl -X "POST" -G "http://localhost:7201/api/v1/query_range" \
  -d "query=third_avenue > 6000" \
  -d "start=$(date -v -45S "+%s")" \
  -d "end=$( date +%s )" \
  -d "step=5s" | jq .
```
```

## Contributing 

You can ask questions and give feedback in the following ways:
- [Create a GitHub issue](https://github.com/m3db/m3/issues)
- [In the public M3 Slack](http://bit.ly/m3slack)
- [In the M3 forum (Google Group)](https://groups.google.com/forum/#!forum/m3db)

M3 welcomes pull requests…

> Read the development guide to help you get setup for building and contributing to M3.

Reviewer 2: ```suggestion
Read [the development guide](DEVELOPMENT.md) to help you get setup for building and contributing to M3.
```

[Distributed TSDB](https://docs.m3db.io/m3db/) and [Query Engine](https://docs.m3db.io/how_to/query/), [Prometheus Sidecar](https://docs.m3db.io/integrations/prometheus/), [Metrics Aggregator](https://docs.m3db.io/overview/components/#m3-aggregator), and more such as [Graphite storage and query engine](https://docs.m3db.io/integrations/graphite/).

## Table of Contents:
- [Install](#install)
- [Dependencies](#dependencies)
- [Usage](#usage)
- [Contributing](#contributing)
- [Community meetings](#community-meetings)

> - [Office Hours](#office-hours)

Reviewer 3: Try to be consistent with capitalizing headers

Reviewer 2: @benraskin92 Good call, and something in the style guide anyway

## More information

- [Documentation](https://docs.m3db.io/)
- [Developer Guide: Getting Started](https://github.com/m3db/m3/blob/master/DEVELOPMENT.md)
- [Slack](http://bit.ly/m3slack)
- [Forum (Google Group)](https://groups.google.com/forum/#!forum/m3db)

## Install 

> ### Dependencies 

Reviewer 3: Should we mention something about `jq` here?

Reviewer 2: Good catch!

The simplest and quickest way to try M3 is to use Docker, read [the M3 quickstart section](https://m3db.io/docs/quickstart) for other options. 

```

## Usage 

The below is a simplified version of the [M3 quickstart guide](https://m3db.io/docs/quickstart/docker/), and we suggest you read that for more details.

1. Start a container

> ```shell

Reviewer 3: should this be `shell` or `bash`?

Reviewer 2: Shell is better if it doesn't matter as bash isn't used everywhere, and actually not the default shell on a lot of environments now, shell can cover bash, zsh, powershelgl etc

docker run -p 7201:7201 -p 7203:7203 --name m3db -v $(pwd)/m3db_data:/var/lib/m3db quay.io/m3db/m3dbnode:v1.0.0
```

2. Create a placement and namespace
```shell
#!/bin/bash
curl -X POST http://localhost:7201/api/v1/database/create -d '{
  "type": "local",
  "namespaceName": "default",
  "retentionTime": "12h"
}' | jq .
```

> 3. Ready a namespace

Reviewer 3: Maybe mark the namespace as ready?

Reviewer 2: This is what it's called in the quickstarts right now, we could change it everywhere later I guess

```shell
curl -X POST http://localhost:7201/api/v1/services/m3db/namespace/ready -d '{
  "name": "default"
}' | jq .
```

> 4. Write metrics

Reviewer 3: 'using the json write endpoint'?

Reviewer 2: Again this is from all the quickstarts, can update later

```shell
#!/bin/bash
curl -X POST http://localhost:7201/api/v1/json/write -d '{
  "tags": 
    {
      "__name__": "third_avenue",
      "city": "new_york",
      "checkout": "1"
    },
    "timestamp": '\"$(date "+%s")\"',
    "value": 3347.26
}'
```

5. Query results
**Linux**
```shell
curl -X "POST" -G "http://localhost:7201/api/v1/query_range" \
  -d "query=third_avenue" \
  -d "start=$(date "+%s" -d "45 seconds ago")" \
  -d "end=$( date +%s )" \
  -d "step=5s" | jq .  
```
**macOS/BSD**
```shell
curl -X "POST" -G "http://localhost:7201/api/v1/query_range" \
  -d "query=third_avenue > 6000" \
  -d "start=$(date -v -45S "+%s")" \
  -d "end=$( date +%s )" \

>   -d "step=5s" | jq .

Reviewer 3: Think you need to close the ``` here

> ## Contributing 

Reviewer 3: newline

You can ask questions and give feedback in the following ways:
- [Create a GitHub issue](https://github.com/m3db/m3/issues)
- [In the public M3 Slack](http://bit.ly/m3slack)
- [In the M3 forum (Google Group)](https://groups.google.com/forum/#!forum/m3db)

M3 welcomes pull requests…
Read [the development guide](DEVELOPMENT.md) to help you get setup for building and contributing to M3.

## Community Meetings

> M3 contributors and maintainers have monthly (every four weeks) meetings. Join our M3 meetup group to receive notifications on upcoming meetings: 

Reviewer 3: Can prob remove `every four weeks`

[https://www.meetup.com/M3-Community/](https://www.meetup.com/M3-Community/).

Recordings of past meetups can be found here: [https://vimeo.com/user/120001164/folder/2290331](https://vimeo.com/user/120001164/folder/2290331).

## Office Hours

> Members of the M3 team will hold office hours on the third Thursday of every month from 11-1pm EST. In order to join, make sure to sign up for a slot here: [https://calendly.com/chronosphere-intro/m3-community-office-hours](https://calendly.com/chronosphere-intro/m3-community-office-hours).

Reviewer 3: `s/team will hold/team hold`

Reviewer 2: Another good style guide catch 😁
